src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const tmpState = {};
+
+  Object.assign(tmpState, state);
+
+  const arrRes = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addP(tmpState, action.extraData);
+
+        const obj1 = {};
+
+        Object.assign(obj1, tmpState);
+        arrRes.push(obj1);
```

Comment:
this logic is reusable, so pls extract it to separate func or const and reuse here

Reply:
I don't quite understand ( could you explaine more detail ?

Reply:
if I right understand...)